cleverhans/max_confidence.py

```diff
@@ -0,0 +1,106 @@
+"""
+The MaxConfidence attack.
+
+An attack designed for use against models that use confidence thresholding
+as a defense.
+If the underlying optimizer is optimal, this attack procedure gives the
+optimal failure rate for every confidence threshold t > 0.5.
+
+Publication: https://openreview.net/forum?id=H1g0piA9tQ
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from six.moves import range
+import tensorflow as tf
+
+from cleverhans.attacks import Attack
+from cleverhans.model import Model
+from cleverhans.attacks import ProjectedGradientDescent
+
+
+class MaxConfidence(Attack):
+  """
+  Initializes with noise and runs iterative optimizer on all classes
+  separately, then chooses the wrong class that had highest confidence.
+
+  Publication: https://openreview.net/forum?id=H1g0piA9tQ
+  """
+
+  def __init__(self, model, sess=None):
+    if not isinstance(model, Model):
+      raise TypeError("Model must be cleverhans.model.Model, got " +
+                      str(type(model)))
+
+    super(MaxConfidence, self).__init__(model, sess)
+    self.base_attacker = ProjectedGradientDescent(model, sess=sess)
+    self.structural_kwargs = self.base_attacker.structural_kwargs
+    self.feedable_kwargs = self.feedable_kwargs
```

Comment:
Is this a typo for `self.base_attacker.feedable_kwargs`?

Reply:
Yes. I guess I must have been running the .pyc the last time I ran this.